File: src/main/java/com/prototype/architecture/layer_02_application/user/create/CreateUserOutput.java
package com.prototype.architecture.layer_02_application.user.create;

import com.prototype.architecture.layer_03_domain.user.UserEntity;
import com.prototype.architecture.layer_03_domain.user.UserId;
import com.prototype.architecture.layer_03_domain.user.UserName;

public record CreateUserOutput(
        UserId userId,
        UserName userName
){

    public static CreateUserOutput of(UserEntity user) {
        return new CreateUserOutput(
                user.getUserId(),
                user.getUserName()
        );
    }

}

File: src/main/kotlin/com/prototype/architecture/layer_02_application/user/create/CreateUserOutput.kt
package com.prototype.architecture.layer_02_application.user.create

import com.prototype.architecture.layer_03_domain.user.UserEntity
import com.prototype.architecture.layer_03_domain.user.UserId
import com.prototype.architecture.layer_03_domain.user.UserName

data class CreateUserOutput(
    val userId: UserId,
    val userName: UserName
) {

    companion object {
        fun of(user: UserEntity): CreateUserOutput {
            return CreateUserOutput(
                user.userId!!,
                user.userName
            )
        }
    }
}